server/build_event_protocol/build_event_proxy/build_event_proxy.go

```diff
@@ -80,7 +80,11 @@ func (c *BuildEventProxyClient) newAsyncStreamProxy(ctx context.Context, opts ..
 			return
 		}
 		asp.PublishBuildEvent_PublishBuildToolEventStreamClient = stream
-		for req := range asp.events {
```

Comment:
nit: could you add a comment about this one? It's kind of interesting :) 

// Iterating over a channel copies the values in that channel, and
// copies of protos are not permitted, so rather than range over the
// channel we read from the channel inside of an outer loop.


or something like that?

Reply:
of course!